apps/frontend/src/components/ChessBoard.tsx

```diff
@@ -305,30 +322,53 @@ export const ChessBoard = memo(({
                                 to: squareRepresentation,
                               });
                             }
+                            const timeTaken =
+                              time - new Date(myMoveStartTime).getTime();
```

Comment:
Will this be the same on both the side?
Should we wait for some ack from the server before setting this?